### PR TITLE
split using simplematch functionality not regex

### LINKED ISSUE #7806: Import-DbaCsv 's automap functionality doesn't work with pipe delimited files going into existing tables
### Verified issue does not already exist?

Yes

### What error did you receive?

`Couldn't auto create ColumnMap :(`

Followed by: 

`File: Encounter_SubServiceLine.txt error message: Error connecting to
[myserver]: Input string was not in a correct format.`

### Steps to Reproduce

```powershell
PS > $delimiter = "|"
PS > $firstline = "ServiceLineCode|SubServiceLineCode|SubServiceDescription"
PS > $firstline -split $Delimiter
```
Returns one row per character.  For automap functionality, we want it to return one row per delimiter + 1

### Are you running the latest release?

Yes

### Other details or mentions

In looking into why we couldn't auto create ColumnMap, with PR #7632 , we reintroduced #6763 [here](https://github.com/sqlcollaborative/dbatools/blob/be7e1f0cbe25530c01c2514f1baa4d4e72204661/functions/Import-DbaCsv.ps1#L557).

I will submit a PR with a fix and I'll work on a test to prevent reintroduction of this in the future <-- the test will be novel work for me, so I may ask for some help, and we'll definitely want to squash my commits.

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.17763.1490                                                                          
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.17763.1490                                                                         
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1                                                                                 

### SQL Server Edition and Build number

Microsoft SQL Server 2019 (RTM-CU12) (KB5004524) - 15.0.4153.1 (X64)   Jul 19 2021 15:37:34   Copyright (C) 2019 Microsoft Corporation  Enterprise Edition (64-bit) on Windows Server 2019 Standard 10.0 <X64> (Build 17763: ) (Hypervisor)

### PR DESCRIPTION
**Pull request by MarkIannucci**
## Type of Change
 - [X] Bug fix (non-breaking change, fixes #7806 )
 - [X] Adding code coverage to existing functionality
 - [X] Pester test is included
 - [x] If new file reference added for test, has is been added to appveyor-lab? yes, once sqlcollaborative/appveyor-lab#10 is complete

### Purpose
Fixes #7806 , increases test coverage

### Approach
Use the "SimpleMatch" option parameter with the -split operator.

### Screenshots
image

### CODE REVIEW

File: tests/Import-DbaCsv.Tests.ps1 [hunk 1]
```diff
@@ -22,6 +22,7 @@ Describe "$CommandName Integration Tests" -Tag "IntegrationTests" {
     $CommaSeparatedWithHeader = "$script:appveyorlabrepo\csv\CommaSeparatedWithHeader.csv"
     $col1 = "$script:appveyorlabrepo\csv\cols.csv"
     $col2 = "$script:appveyorlabrepo\csv\col2.csv"
+    $pipe3 = "$script:appveyorlabrepo\csv\pipe3.psv"
```
Comment (wsmelton) [inline tests/Import-DbaCsv.Tests.ps1:25]: This file would have to be added with example data in order for it to be usable.

File: tests/Import-DbaCsv.Tests.ps1 [hunk 2]
```diff
@@ -30,9 +31,10 @@ Describe "$CommandName Integration Tests" -Tag "IntegrationTests" {
             $results | Should -Be $null
         }
 
-        It "creates the right columnmap (#7630)" {
+        It "creates the right columnmap (#7630), handles pipe delimiters (#7806)" {
             $null = Import-DbaCsv -SqlInstance $script:instance1 -Path $col1 -Database tempdb -AutoCreateTable -Table cols
             $null = Import-DbaCsv -SqlInstance $script:instance1 -Path $col2 -Database tempdb -Table cols
+            $null = Import-DbaCsv -SqlInstance $script:instance1 -Path $pipe3 -Database tempdb -Table cols -Delimiter "|"
```
Comment (wsmelton) [inline tests/Import-DbaCsv.Tests.ps1:37]: This is a false positive test considering `$pipe3` has no content from the Appveyor run.

File: tests/Import-DbaCsv.Tests.ps1 (hunk 1 above)
Comment (wsmelton) [reply tests/Import-DbaCsv.Tests.ps1:25]: The file needs to be added to the [appveyor-lab/csv](https://github.com/sqlcollaborative/appveyor-lab/tree/master/csv) folder.

**Review by wsmelton — CHANGES_REQUESTED** [review]

File: tests/Import-DbaCsv.Tests.ps1 (hunk 1 above)
Comment (potatoqualitee) [reply tests/Import-DbaCsv.Tests.ps1:25]: sweet, @MarkIannucci submitted a PR and I accepted

File: tests/Import-DbaCsv.Tests.ps1 (hunk 2 above)
Comment (potatoqualitee) [reply tests/Import-DbaCsv.Tests.ps1:37]: also true, i'll rerun the test but it seems it is indeed a false positive

**Comment by potatoqualitee** [comment]
Thanks so much! Looking forward to merging once the test is fixed.

**Comment by potatoqualitee** [comment]
added a new test but still baffled as to why `Import-DbaCsv -SqlInstance $script:instance1 -Path $pipe3 -Database tempdb -Table cols2 -Delimiter "|"` just didn't throw.

the new test tests a new table, let's see how it goes!

**Comment by potatoqualitee** [comment]
awesome, works!

**Pull request merged**